render-test/manifest_parser.cpp

```diff
@@ -412,6 +412,18 @@ mbgl::optional<Manifest> ManifestParser::parseManifest(const std::string& manife
         }
     }
 
+    auto filter = testFilter;
+    if (document.HasMember("filter") && filter.empty()) {
```

Comment:
nit: ` if ( filter.empty() && document.HasMember("filter"))` would be a bit faster :)